File: Java/13-Method-Overloading.java
// 1. Write two methods with the same name but different number of parameters of same type and call the methods from main method

class OverloadExample1 {
    void display(int a) {
        System.out.println("One parameter: " + a);
    }

    void display(int a, int b) {
        System.out.println("Two parameters: " + a + ", " + b);
    }

    public static void main(String[] args) {
        OverloadExample1 obj = new OverloadExample1();
        obj.display(10);
        obj.display(10, 20);
    }
}






// 2. Write two methods with the same name but different number of parameters of different data type 
//    and call the methods from main method

class OverloadExample2 {
    void show(int a) {
        System.out.println("Integer: " + a);
    }

    void show(String b) {
        System.out.println("String: " + b);
    }

    public static void main(String[] args) {
        OverloadExample2 obj = new OverloadExample2();
        obj.show(100);
        obj.show("Java");
    }
}








// 3. Write two methods with the same name and same number of parameters of same type and call from main method










// 4. Write two methods with the same name and same number of parameters of different type and call from main method

class OverloadExample4 {
    void display(int a) {
        System.out.println("Integer: " + a);
    }

    void display(double b) {
        System.out.println("Double: " + b);
    }

    public static void main(String[] args) {
        OverloadExample4 obj = new OverloadExample4();
        obj.display(10);
        obj.display(3.14);
    }
}

File: Python/13-Method-Overloading.py
# 1. Write two methods with the same name but different number of parameters of same type and call the methods
def greet(name):
    print(f"Hello, {name}!")

def greet(name, age):
    print(f"Hello, {name}! You are {age} years old.")

greet("Alice")
greet("Bob", 25)





# 2. Write two methods with the same name but different number of parameters of different data type and call the methods
def display_info(name, age):
    print(f"Name: {name}, Age: {age}")

def display_info(name, age, is_student):
    print(f"Name: {name}, Age: {age}, Student: {is_student}")

display_info("Alice", 25)
display_info("Bob", 30, True)





# 3. Write two methods with the same name and same number of parameters of same type
def add(x, y):
    return x + y

def add(x, y):
    return x * y

print(add(5, 10)) 
